app/models/source.rb

```diff
@@ -141,5 +141,11 @@ def display_name
       #   ex.: Sources::DigitalOcean.display_name == "Digital Ocean"
       self.name.underscore.split(%r[/]).last.split(%r[_]).collect(&:capitalize).join ' '
     end
+    
+    def valid_source_types
```

Comment:
This method introduces a cyclical dependency between Source and its child classes.  Create some independent class/module to house this method instead.